src/xray/askui-xray-step-reporter.ts

```diff
@@ -0,0 +1,155 @@
+import { Reporter, Step, StepStatus, ReporterConfig } from 'askui';
+import { convertPngDataUrlToBase64 } from '../utils/image-reporting-utils';
+import path from 'path';
+import fs from 'fs';
+
+/*
+ * Possible XRay status: PASS, FAIL, TODO
+ */
+enum Status {
+  PASS = 'PASS',
+  FAIL = 'FAIL',
+  TODO = 'TODO',
+}
+
+type XRayTestObject = {
+  testKey: string;
+  status?: Status;
+  start?: string;
+  finish?: string;
+  steps?: XRayStep[];
+}
+
+type XRayEvidence = {
+  data: string;
+  filename: string;
+  contentType: string;
+}
+
+type XRayStep = {
+  status?: Status;
+  actualResult?: string;
+  evidences: XRayEvidence[];
+}
```

Comment:
Well doen putting this into types ;) Use interfaces instead of raw types wherever possible as they have more constraints so that it they are easier and more powerful to work with ;) (Here it is always possible as far as I can see)